lib/poseidon/sync_producer.rb

```diff
@@ -42,7 +42,12 @@ def send_messages(messages)
       messages_to_send = MessagesToSend.new(messages, @cluster_metadata)
 
       if refresh_interval_elapsed?
-        refresh_metadata(messages_to_send.topic_set)
+        begin
+          refresh_metadata(messages_to_send.topic_set)
+        rescue Errors::UnableToFetchMetadata => e
```

Comment:
I have no idea how prevalent the situation would be, so I defer to you on that and trust that you're correct.

I was actually thinking, in lieu of a backoff, using andys CircuitBreaker gem to handle this kind of thing - as it seems to be exactly the kind of thing it's designed for. 

But that said, it's probably more work than it's worth atm, and so something like only checking the metadata if it's been X seconds since our last attempt && it was a failed last attempt, isn't a terrible middle ground.